### PR TITLE
Add c-nav-icon and c-nav-text to build.styl

### PR DESCRIPTION
**Pull request by CPatchane**

### CODE REVIEW

File: stylus/cozy-ui/build.styl [hunk 1]
```diff
@@ -249,6 +255,12 @@
 .c-nav-link
     @extend $nav-link
 
+.c-nav-icon
```
Comment (CPatchane) [inline stylus/cozy-ui/build.styl]: Do you see an example to put for `.c-nav-icon` in the comments above?

Comment (GoOz) [reply stylus/cozy-ui/build.styl]: Not really I haven't use this component yet, but I guess you can get inspiration from the react styleguide.

Comment (CPatchane) [reply stylus/cozy-ui/build.styl]: I will keep it out of the comment for now, I use it with the `<Icon/>` react component. We may need to test it with a simple `img` or `svg`

**Comment by CPatchane** [comment]
I make a better usage of `c-nav-icon`. Tested with the NavIcon component as well and it works.
I add an example in the kss styleguide @GoOz

File: stylus/cozy-ui/build.styl (hunk 1 above)
Comment (CPatchane) [reply stylus/cozy-ui/build.styl]: Tested with a svg here and it works :)

File: stylus/components/nav.styl [hunk 2]
```diff
@@ -91,7 +91,7 @@ $nav-link
         box-shadow   inset 4px 0 0 0 dodgerBlue
         font-weight  bold
 
-        .nav-link__icon
```
Comment (CPatchane) [inline stylus/components/nav.styl]: @GoOz Should we need a retro-compatibility here?

Comment (GoOz) [reply stylus/components/nav.styl]: Well, ask @ptbrowne I never used it myself yet.

Comment (CPatchane) [reply stylus/components/nav.styl]: Ok, since it was used only for the `<Nav />` component indeed, we don't need one.

**Pull request merged**